src/internal/packager2/layout/import.go

```diff
@@ -0,0 +1,408 @@
+	}
+
+	name := component.Name
+	if component.Import.Name != "" {
+		name = component.Import.Name
+	}
+
+	absCachePath, err := config.GetAbsCachePath()
+	if err != nil {
+		return "", err
+	}
+	cache := filepath.Join(absCachePath, "oci")
+	if err := helpers.CreateDirectory(cache, helpers.ReadWriteExecuteUser); err != nil {
+		return "", err
+	}
+
+	// Get the descriptor for the component.
+	remote, err := zoci.NewRemote(component.Import.URL, zoci.PlatformForSkeleton())
+	if err != nil {
+		return "", err
+	}
+	_, err = remote.ResolveRoot(ctx)
+	if err != nil {
+		return "", fmt.Errorf("published skeleton package for %s does not exist: %w", component.Import.URL, err)
+	}
+	manifest, err := remote.FetchRoot(ctx)
+	if err != nil {
+		return "", err
+	}
+	componentDesc := manifest.Locate(filepath.Join(layout.ComponentsDir, fmt.Sprintf("%s.tar", name)))
+	if oci.IsEmptyDescriptor(componentDesc) {
+		return "", fmt.Errorf("component %s not found", name)
+	}
+
+	store, err := ocistore.New(cache)
+	if err != nil {
+		return "", err
+	}
+	exists, err := store.Exists(ctx, componentDesc)
+	if err != nil {
+		return "", err
+	}
+	if !exists {
+		err = remote.CopyToTarget(ctx, []ocispec.Descriptor{componentDesc}, store, remote.GetDefaultCopyOpts())
+		if err != nil {
+			return "", err
+		}
+	}
+	dir := filepath.Join(cache, "dirs", componentDesc.Digest.Encoded())
+	if err := helpers.CreateDirectory(dir, helpers.ReadWriteExecuteUser); err != nil {
+		return "", err
+	}
+	tu := archiver.Tar{
+		OverwriteExisting: true,
+		// removes /<component-name>/ from the paths
+		StripComponents: 1,
+	}
+	tb := filepath.Join(cache, "blobs", "sha256", componentDesc.Digest.Encoded())
+	err = tu.Unarchive(tb, dir)
+	if err != nil {
+		return "", err
+	}
+	abs, err := filepath.Abs(packagePath)
+	if err != nil {
+		return "", err
+	}
+	rel, err := filepath.Rel(abs, dir)
+	if err != nil {
+		return "", err
+	}
+	return rel, nil
+}
+
+func overrideComponent(c v1alpha1.ZarfComponent, override v1alpha1.ZarfComponent) (v1alpha1.ZarfComponent, error) {
+	// Metadata
+	c.Name = override.Name
+	c.Default = override.Default
+	c.Required = override.Required
+
+	// Override description if it was provided.
+	if override.Description != "" {
+		c.Description = override.Description
+	}
+
+	if override.Only.LocalOS != "" {
+		if c.Only.LocalOS != "" {
+			return v1alpha1.ZarfComponent{}, fmt.Errorf("component %q: \"only.localOS\" %q cannot be redefined as %q during compose", c.Name, c.Only.LocalOS, override.Only.LocalOS)
+		}
+		c.Only.LocalOS = override.Only.LocalOS
+	}
+
+	// Deprecated
+	// Override cosign key path if it was provided.
+	if override.DeprecatedCosignKeyPath != "" {
+		c.DeprecatedCosignKeyPath = override.DeprecatedCosignKeyPath
+	}
+
+	c.DeprecatedGroup = override.DeprecatedGroup
+
+	// Merge deprecated scripts for backwards compatibility with older zarf binaries.
+	c.DeprecatedScripts.Before = append(c.DeprecatedScripts.Before, override.DeprecatedScripts.Before...)
```

Comment:
small nit: i think `overrideDeprecated` and `overrideActions`  was easier to consume as two functions